### PR TITLE
fix: intel compute runtime packages

### PR DESCRIPTION
**Pull request by bo0tzz**
To mirror https://github.com/immich-app/immich/pull/11818

### CODE REVIEW

**Review by mertalev — CHANGES_REQUESTED** [review]
This is removing other dependencies we also get from sid. It should only remove `intel-opencl-icd` from the command.

**Review by mertalev — DISMISSED** [review]

File: server/bin/configure-apt.sh [hunk 1]
```diff
@@ -3,11 +3,11 @@
...
 
 sed -i -e's/ main/ main contrib non-free non-free-firmware/g' /etc/apt/sources.list.d/debian.sources
-sed -i -e's/ bookworm-updates/ bookworm-updates sid/g' /etc/apt/sources.list.d/debian.sources
+sed -i -e's/ bookworm-updates/ bookworm-updates trixie/g' /etc/apt/sources.list.d/debian.sources
 
 # default priority is 500, so we set unstable to 450 to prefer stable packages
 cat > /etc/apt/preferences.d/preferences << EOL
 Package: *
-Pin: release a=unstable
+Pin: release n=trixie
```
Comment (martabal) [inline server/bin/configure-apt.sh:11]: Should be `a=testing` (#105)

File: server/Dockerfile [hunk 2]
```diff
@@ -104,13 +104,22 @@ RUN ./configure-apt.sh && \
...
+    RUN if [ $(arch) = "x86_64" ]; then \
+      wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.17193.4/intel-igc-core_1.0.17193.4_amd64.deb && \
+      wget https://github.com/intel/intel-graphics-compiler/releases/download/igc-1.0.17193.4/intel-igc-opencl_1.0.17193.4_amd64.deb && \
+      wget https://github.com/intel/compute-runtime/releases/download/24.26.30049.6/intel-opencl-icd-dbgsym_24.26.30049.6_amd64.ddeb && \
+      wget https://github.com/intel/compute-runtime/releases/download/24.26.30049.6/intel-opencl-icd_24.26.30049.6_amd64.deb && \
+      wget https://github.com/intel/compute-runtime/releases/download/24.26.30049.6/libigdgmm12_22.3.20_amd64.deb && \
+      dpkg -i *.deb && \
+      apt-get install -t testing --no-install-recommends -yqq intel-media-va-driver-non-free && \
+      rm *.deb; \
+    fi && \
```
Comment (martabal) [inline server/Dockerfile:122]: I don't really like it, it's hard to tell if there are any missing dependencies.

I have something like this for the AIO image, which comes from linuxserver:
```bash
  if [ $(arch) = "x86_64" ]; then \
    echo "**** install intel dependencies ****" && \
    apt-get install --no-install-recommends -y \
      intel-media-va-driver-non-free \
      ocl-icd-libopencl1 && \
    INTEL_DEPENDENCIES=$(curl -sX GET "https://api.github.com/repos/intel/compute-runtime/releases/latest" | jq -r '.body' | grep wget | grep -v .sum | grep -v .ddeb | sed 's|wget ||g') && \
    mkdir -p /tmp/intel && \
    for i in $INTEL_DEPENDENCIES; do \
      curl -fS --retry 3 --retry-connrefused -o \
        /tmp/intel/$(basename "${i%$'\r'}") -L \
        "${i%$'\r'}"; \
    done && \
    dpkg -i /tmp/intel/*.deb; \
  fi && \
```

Not sure if we need `wget https://github.com/intel/compute-runtime/releases/download/24.26.30049.6/intel-opencl-icd-dbgsym_24.26.30049.6_amd64.ddeb`, it's not installed by `dpkg -i *.deb`

Comment (mertalev) [reply server/Dockerfile:122]: Interesting! I guess an advantage of this is that you can just point it at a release (or latest) without specifying the specific file name of each dependency.

Comment (zackpollard) [reply server/Dockerfile:122]: I think the issue is we don't want all of the dependencies from the release, maybe we could filter which ones we take though

Comment (zackpollard) [reply server/Dockerfile]: Removed the unused ddeb package, leaving the rest as-is for now

**Review by mertalev — APPROVED** [review]

**Pull request merged**